### PR TITLE
fix up Mortality and observer

### PR DESCRIPTION
**Pull request by stevebachmeier**
### Description

- *Category*:  refactor
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5038

### Changes and notes
 
Same as other subtasks, modifies the component and its observer
to utilize the new results processing and recent refactors.

### Testing

Tests pass

### CODE REVIEW

File: src/vivarium_public_health/metrics/mortality.py [hunk 1]
```diff
@@ -142,3 +150,22 @@ def count_deaths(self, x: pd.DataFrame) -> float:
...
+    def write_mortality_results(
+        self, cause: str, measure: str, results: pd.DataFrame
+    ) -> None:
+        measure_name = measure.split("_due_to_")[0]
+        write_dataframe_to_parquet(
+            results=results,
+            measure=measure_name,
+            entity_type="cause",
+            entity=cause,
+            sub_entity=None,  # FIXME
```
Comment (stevebachmeier) [inline src/vivarium_public_health/metrics/mortality.py:167]: Is this actually correct? Or can there be a subentity like implied in the [RT design doc (see second row)](https://vivarium-research.readthedocs.io/en/latest/model_design/vivarium_features/standard_results_formats/index.html#some-examples)

File: tests/metrics/test_mortality_observer.py [hunk 2]
```diff
@@ -77,91 +83,75 @@ def simulation_after_one_step(base_config, base_plugins):
...
     # same test on second time step
     simulation_after_one_step.step()
-    expected_deaths, expected_ylls = get_expected_results(
-        simulation_after_one_step,
-        expected_deaths=expected_deaths,
-        expected_ylls=expected_ylls,
+    expected = get_expected_results(simulation_after_one_step, expected)
+    _assert_metric_correctness(simulation_after_one_step, expected)
+
+    # ensure actual output files match as well
```
Comment (stevebachmeier) [inline tests/metrics/test_mortality_observer.py:136]: This is a little silly. We typically have been testing correctness against the output/written datasets but I didn't want to refactor everything here so I kept w/ comparing it to the metrics pipeline. But then I did want to compare the written results as well so here we are

File: tests/metrics/test_categorical_risk_observer.py [hunk 3]
```diff
@@ -105,6 +105,25 @@ def test_observation_correctness(base_config, simulation_after_one_step, categor
     assert set(file.name for file in results_files) == set(["person_time_test_risk.parquet"])
     results = pd.read_parquet(results_files[0])
 
+    # Check columns
```
Comment (stevebachmeier) [inline tests/metrics/test_categorical_risk_observer.py:108]: I just forgot to add this check to the risk observer in the previous PR

**Review by rmudambi — APPROVED** [review]

File: src/vivarium_public_health/metrics/mortality.py [hunk 4]
```diff
@@ -69,6 +62,16 @@ def __init__(self):
     # Properties #
     ##############
 
+    @property
+    def configuration_defaults(self) -> Dict[str, Any]:
+        """
+        A dictionary containing the defaults for any configurations managed by
+        this component.
+        """
+        config_defaults = super().configuration_defaults.copy()
```
[the remark this answers is not in the document]
Comment (stevebachmeier) [reply src/vivarium_public_health/metrics/mortality.py:71]: leftover from when I was trying to use `.update` but changed strategy.  will remove

File: src/vivarium_public_health/metrics/mortality.py (hunk 1 above)
Comment (stevebachmeier) [reply src/vivarium_public_health/metrics/mortality.py:167]: @rmudambi But then what goes into entity?

Comment (stevebachmeier) [reply src/vivarium_public_health/metrics/mortality.py:167]: The RT design doc has the following example for deaths:
image

I feel like we're missing Disease information (i.e. "infected" state) at this point, though

Comment (stevebachmeier) [reply src/vivarium_public_health/metrics/mortality.py:167]: On wait - I think I see what I need to do

File: tests/metrics/test_mortality_observer.py [hunk 5]
```diff
@@ -196,14 +210,36 @@ def test_aggregation_configuration(base_config, base_plugins):
     aggregate_sim._data.write("cause.all_causes.cause_specific_mortality_rate", acmr_data)
     aggregate_sim.setup()
     aggregate_sim.step()
-    results = aggregate_sim.get_value("metrics")
-    pop = aggregate_sim.get_population()
+    aggregate_sim.finalize()
```
Comment (patricktnast) [inline tests/metrics/test_mortality_observer.py:213]: you do ln 213-222 several times. Does it make sense to extract it?

**Review by patricktnast — APPROVED** [review]

File: src/vivarium_public_health/metrics/mortality.py [hunk 6]
```diff
@@ -62,13 +61,17 @@ def __init__(self):
     # Properties #
     ##############
 
+    @property
+    def disease_classes(self) -> List:
```
Comment (rmudambi) [inline src/vivarium_public_health/metrics/mortality.py:65]: A better name might be `mortality_classes`

**Review by rmudambi — APPROVED** [review]

**Pull request merged**